cmd/virsnap/export.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2019 The virnsnap authors. See file "AUTHORS".
+// Licensed under the MIT License. You have obtained a copy of the License at
+// the "LICENSE" file in this repository.
+
+// Package main implements the handlers for the different command line arguments.
+package main
+
+import (
+	"path/filepath"
+
+	"github.com/joroec/virsnap/pkg/fs"
+	"github.com/joroec/virsnap/pkg/virt"
+
+	"github.com/libvirt/libvirt-go"
+	"github.com/spf13/cobra"
+)
+
+var (
+	// outputDir is the target directory of the backup
+	outputDir = ""
+
+	// snapshot determines whether virsnap should make a new snapshot after the
+	// machine was shut down.
+	snapshot = true
+
+	// exportCmd is a global variable defining the corresponding cobra command
+	exportCmd = &cobra.Command{
+		Use:   "export --output-dir <export_directory> <regex1> [<regex2>] [<regex3>] ...",
+		Short: "Export a VM by copying the hard drive images to an output directory",
+		Long: "Export a VM by copying the hard drive images and an copy of the " +
+			"VMs XML descriptor file to an output directory. Exports any found " +
+			"virtual machine with a name matching at least one of the given " +
+			"regular expressions. To ensure a safe export, the VM needs to be " +
+			"shutoff. Hence, virsnap shuts down the VM if its running, exports the " +
+			"disk files and restores the VM's previous state afterwards. Apart from " +
+			"this, there is an option to create a snapshot of the VM after " +
+			"shutdowning and before exporting to the given directory.",
+		Args: cobra.MinimumNArgs(1),
+		Run:  exportRun,
+	}
+)
+
+// init is a special golang function that is called exactly once regardless
+// how often the package is imported.
+func init() {
+	// initialize flags and arguments needed for this command
+	exportCmd.Flags().StringVarP(&outputDir, "output-dir", "o", "default?",
+		"desc")
+	exportCmd.MarkFlagRequired("output-dir")
+
+	exportCmd.Flags().BoolVarP(&snapshot, "snapshot", "s", true, "Make new "+
+		"snapshot after shutdowning the machine.")
+
+	exportCmd.Flags().IntVarP(&timeout, "timeout", "t", 3, "Timeout in minutes "+
```

Comment:
I'm a fan of having base units, where possible. Maybe `seconds` is better as a unit?

Reply:
It is consistent to "create" command using minutes as unit. Nevertheless, I agree with you. A general change to base units seems to be desireable in a future rework of the UI.